src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java

```diff
@@ -1,16 +1,25 @@
 package mate.academy.hibernate.relations.service.impl;
 
+import mate.academy.hibernate.relations.dao.CountryDao;
+import mate.academy.hibernate.relations.dao.impl.CountryDaoImpl;
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import mate.academy.hibernate.relations.util.HibernateUtil;
 
 public class CountryServiceImpl implements CountryService {
+    private CountryDao countryDao;
+
+    public CountryServiceImpl() {
+        countryDao = new CountryDaoImpl(HibernateUtil.getSessionFactory());
```

Comment:
Consider passing the `SessionFactory` as a parameter to the constructor of `CountryServiceImpl` instead of directly accessing `HibernateUtil.getSessionFactory()`. This approach improves testability and decouples the service from the utility class.